src/DatadirTestsFromDirectoryProvider.php

```diff
@@ -67,11 +67,24 @@ protected function processOneTest(SplFileInfo $testSuite): void
         $sourceDatadirDirectory = $workingDirectory . '/source/data';
         $expectedStdout = null;
         $expectedStderr = null;
+        $expectedReturnCodeFile = $workingDirectory . '/expected-code';
         $expectedReturnCode = null;
         $expectedOutputDirectory = null;
         $outTemplateDir = $workingDirectory . '/expected/data/out';
+
+        if (file_exists($expectedReturnCodeFile)) {
+            $returnCode = trim(file_get_contents($expectedReturnCodeFile));
+            if (preg_match('~^[012]$~', $returnCode)) {
+                $expectedReturnCode = (int) $returnCode;
+            } else {
+                throw new \InvalidArgumentException($name . ': Expecting invalid return code. Possible codes are: 0, 1, 2.');
```

Comment:
Ideálně vždycky exceptiony se sprintf a vyplnit do nich "vadnou" hodnotu. Tj. `sprintf('%s: Expecting invalid return code "%s". Possible codes are ...', $name, $expectedReturnCode)`

Reply:
Když už jsem přidával ty testy, tak jsem to rovnou opravil.